File: 201. Bitwise AND of Numbers Range.cpp
#include <bitset>
class Solution {
public:
    int rangeBitwiseAnd(int m, int n) {
        string bin_m = bitset <32> (m).to_string();
        string bin_n = bitset <32> (n).to_string();
        int res = 0;
        for (int i = bin_m.length()-1; i >= 0; i--)
        {
            bool equal = true;
            for (int j = 0; j <= i; j++)
            {
                if (bin_m[j] != bin_n[j])
                {
                    equal = false;
                    break;
                }
            }
            if (equal)
            {
                for (int j = 0; j <= i; j++)
                    res += (bin_m[j]-'0') * pow(2, bin_m.size() - 1 - j);
                break;
            }
        }
        return res;
    }
};

File: 201. Bitwise AND of Numbers Range.py
class Solution(object):
    def rangeBitwiseAnd(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        bin_m = bin(m)[2:]
        bin_n = bin(n)[2:]
        for i in xrange(len(bin_n) - len(bin_m)):
            bin_n = '0' + bin_n
        res = 0
        for i in xrange(len(bin_n), -1, -1):
            if bin_m[:i] == bin_n[:i]:
                for j in xrange(i-1, -1, -1):
                    res += int(bin_m[j]) * pow(2, len(bin_m) - 1 - j)
                break
        return res
